docs/aurora/data-science/jupyter.md

```diff
@@ -0,0 +1,136 @@
+   Open a new terminal tab or window on your local machine and run the following command:
+   ```bash
+   ssh -L 9999:127.0.0.1:9999 <your-username>@aurora.alcf.anl.gov
+   ```
+   - Replace `9999` with another port if it is unavailable.
+
+4. **Access JupyterLab**:
+   Open your browser and navigate to the address copied above:
+   ```
+   http://localhost:9999/?token=<provided-token>
+   ```
+
+---
+
+## 3. Run JupyterLab on a Compute Node
+
+### Step 1: Request a Compute Node
+You need a job running on Aurora to launch JupyterLab on a compute node. Below is an example of how to submit an interactive job to request a compute node. Note that you can also connect to one of the compute nodes of any of your non-interactive jobs that is running on Aurora. See [Running Jobs on Aurora](../running-jobs-aurora.md) for more information.
+
+1. **Submit a Job**:
+   Submit an interactive job to request a compute node:
+   ```bash
+   qsub -l select=1 -l walltime=60:00 -A <project_name> -q <queue_name> -I
+   ```
+   You need to modify the `-A` and `-q` options to match your project name and queue name as well as the resources you need.
+
+2. **Find Compute Node Hostname**:
+   Once the interactive job starts, you will be connected to the compute node. You can check the hostname with:
+   ```bash
+   hostname
+   ```
+   If you are not running an interactive job, you can find the hostname of the compute node by checking the `qstat` output.
+   First, you need to find the job ID of the job you are interested in. The following command will list all the jobs you have submitted. You need the ID of any one of the **running** jobs you are interested in.
+   ```bash
+   qstat -u <your_username>
+   ```
+   Then, we can find the hostname of the compute node by running:
+   ```bash
+   qstat -f <job_id>
+   ```
+   The hostname of the compute node will be displayed in the `exec_host` field. You can extract the hostname from the output with the following command:
+   ```bash
+   qstat -f <job_id> | awk -F '=' '/exec_host/ {print $2}' | tr '+' '\n' | cut -d '/' -f 1
+   ```
+   This should give you a hostname like `x4603c0s0b0n0`. Note that you can ssh into the compute nodes only when your job is running.
+
+### Step 2: Start JupyterLab on the Compute Node
+1. **SSH to the Compute Node**:
+   From the login node:
+   ```bash
+   ssh <compute_node_hostname>
+   ```
+2. **Activate the Environment**:
+   ```bash
+   source <path_to_your_virtual_environment>/bin/activate
+   ```
+3. **Start JupyterLab**:
+   ```bash
+   jupyter lab --no-browser --port=9999
+   ```
+
+### Step 3: Set Up SSH Tunneling
+1. **Tunnel from Compute Node to Local Machine**:
+   On your local machine, run:
+   ```bash
+   ssh -L 9999:127.0.0.1:9999 -J <your-username>@aurora.alcf.anl.gov <your-username>@<compute_node_hostname>
+   ```
+   Please note that the `-J` option is used to specify the jump host, which is the Aurora login node.
+   - Replace `9999` with another port if it is unavailable.
```

Comment:
```suggestion
   Replace `9999` with another port if it is unavailable.
```

Reply:
should you mention how to check what ports are in use, etc. via `lsof`?

Reply:
Never had the problem of finding an open port, so didn't think about it.